pir/cpp/client.cpp

```diff
@@ -41,69 +46,69 @@ StatusOr<std::unique_ptr<PIRClient>> PIRClient::Create(
 }
 
 StatusOr<PIRPayload> PIRClient::CreateRequest(std::size_t index) const {
-  if (index >= this->DBSize()) {
+  const auto poly_modulus_degree =
+      context_->Parameters()->GetEncryptionParams().poly_modulus_degree();
+  if (index >= DBSize()) {
     return InvalidArgumentError("invalid index");
   }
-  std::vector<std::int64_t> request(DBSize(), 0);
-  request[index] = 1;
-
-  return this->encryptRequestBuffer(request);
-}
+  if (index >= poly_modulus_degree) {
+    // Not yet implemented
+    return InvalidArgumentError("More than 1 CT needed for selection vector");
+  }
 
-StatusOr<int64_t> PIRClient::ProcessResponse(const PIRPayload& response) const {
-  ASSIGN_OR_RETURN(auto decrypted, this->decryptResponseBuffer(response));
-  std::map<uint64_t, int64_t> result;
+  // Figure out 1 / m so that after oblivious expansion it's just 1 in the
+  // correct location.
+  const auto& plain_mod =
+      context_->Parameters()->GetEncryptionParams().plain_modulus();
+  if (plain_mod.uint64_count() > 1) {
+    return InternalError("Plaintext modulus too big");
+  }
+  uint64_t inverse_m;
+  if (!seal::util::try_invert_uint_mod(next_power_two(DBSize()),
+                                       plain_mod.value(), inverse_m)) {
+    return InternalError("Could not invert m value");
```

Comment:
Is there a way to test the errors too?

Reply:
I'd have to find a value of m that isn't invertible, and to be honest I'm not sure how to do that :)